### PR TITLE
Order Details: sync missing variations in order details

### PR DESCRIPTION
**Pull request by jaclync**
Part of #2167

## Why

Before this PR, we only sync products in order details. Since some order items could be variations instead of products, these order items would never be loaded until the user goes to another screen that syncs the variations (e.g. visiting a variable product's variations page). This PR syncs product variations in order details so that variation order items in shipping label cards are synced for UI display. 

## Changes

- Added `ProductVariationAction.requestMissingVariations` for requesting missing variations in an order in `ProductVariationStore` with test cases
  - Unfortunately, I didn't find an API endpoint to request a list of variations so the store makes an API request for each order item that misses its variation and signals completion when all requests finish. Normally, an order wouldn't have a lot of order items
- Synced missing variations in `OrderDetailsViewModel` and `OrderDetailsViewController` following when `syncProducts` is called

## Testing

Prerequisites: the site has [WooCommerce Shipping & Tax plugin](https://docs.woocommerce.com/document/woocommerce-shipping-and-tax/) and there is an order with at least one non-refunded shipping label and an order with at least one refunded shipping label.

You can set up your store to use a test credit card with instructions in p91TBi-3xD-p2.

- In core, create an order with a variation. Then create a shipping label for this order
- Launch the app, log out and in to the store without going to the Products tab
- Go to the orders tab
- Tap on the order with a variation & shipping label --> after syncing, the variation order item should show its price, image (if any), and total

## Example screenshots 

before | after
-- | --
Simulator Screen Shot - iPhone 11 - 2020-12-03 at 10 51 15 | Simulator Screen Shot - iPhone 11 - 2020-12-03 at 13 55 27

 
Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Yosemite/YosemiteTests/Stores/ProductVariationStoreTests.swift [hunk 1]
```diff
@@ -455,7 +455,7 @@ final class ProductVariationStoreTests: XCTestCase {
 
     /// Verifies that `ProductVariationAction.updateProductVariation` returns an error whenever there is an error response from the backend.
     ///
-    func testUpdatingProductVariationReturnsErrorUponReponseError() {
+    func testUpdatingProductVariationReturnsErrorUponResponseError() {
```
Comment (pmusolino) [inline Yosemite/YosemiteTests/Stores/ProductVariationStoreTests.swift]: Do it worth renaming the method using the underscore?

**Review by pmusolino — APPROVED** [review]
Looks good to me 💯 



**I've only a question**: it seems that the sync of product variations inside an order detail, affect only the products inside the Package, but not the ones inside the `Products` section, as you can see from the screen below. Shouldn't these be updated too?

**Review by Garance91540 — APPROVED** [review]
👏👏👏

**Comment by shiki** [comment]
@jaclync, will this also fix https://github.com/woocommerce/woocommerce-ios/issues/2113?

**Comment by jaclync** [comment]
> Whether this PR fixes the missing variation order item images in the Products section

@pmusolino @shiki Great question! No, this PR does not fix the missing image in the variation order items in the Products section but now I have a better idea to fix #2113. For the order items in the Products section, it currently only tries finding the matching `Product` but not the `ProductVariation` in and the `imageURL` is from the optional `Product`.

https://github.com/woocommerce/woocommerce-ios/blob/97269f4cbdba00015524da51eff5b9dccac81b0c/WooCommerce/Classes/ViewModels/Order%20Details/OrderDetailsDataSource.swift#L552-L554

I'm planning to fix this by finding the matching `ProductVariation` if the order item is associated with a variation. will work on #2113 this sprint!

File: Yosemite/YosemiteTests/Stores/ProductVariationStoreTests.swift (hunk 1 above)
Comment (jaclync) [reply Yosemite/YosemiteTests/Stores/ProductVariationStoreTests.swift]: yup updated in: https://github.com/woocommerce/woocommerce-ios/pull/3278/commits/f436d3e9aefd1ff68f23e1cc89639c70b39fd44f

**Pull request merged**